extensions/wikia/YouTube/YouTube.php

```diff
@@ -96,21 +96,24 @@ function upgradeYouTubeTag( $editpage, $request ) {
 	// <youtube> tags within it.  We don't want to convert anything within <nowiki>
 	$text = preg_replace_callback(
 		'/(<nowiki>.*?<\/nowiki>)|(<youtube([^>]*)>([^<]+)<\/youtube>)/i',
-		function ($matches) {
+		function ( $matches ) {
 			// If we don't have a youtube match (its a nowiki tag) return as is
-			if ( empty($matches[2]) ) {
+			if ( empty( $matches[2] ) ) {
 				return $matches[0];
 			}
 
 			// Separate the Youtube ID and parameters
-			$paramText = trim($matches[3]);
-			$ytid   = $matches[4];
+			$paramText = trim( $matches[3] );
+			// Node value can look like: <youtube_id>|400px|thumb|center
+			// @TODO evaluate calling parser to parse params and upload correctly styled video
+			$nodeValues = explode( '|', $matches[4] );
```

Comment:
Yes. Only when <youtube>%something%</youtube> is present code is going to be executed. In cases like:
* <youtube>%something% - parser won't call this function
* <youtube></youtube> - it won't match regular expression
* <youtube %something%><youtube> - it would match regular expression and 4th matching group would be empty string